core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -98,14 +103,22 @@ class Replica(val brokerId: Int, val topicPartition: TopicPartition) extends Log
    * fetch request is always smaller than the leader's LEO, which can happen if small produce requests are received at
    * high frequency.
    */
-  def updateFetchState(
+  def updateFetchStateOrThrow(
     followerFetchOffsetMetadata: LogOffsetMetadata,
     followerStartOffset: Long,
     followerFetchTimeMs: Long,
     leaderEndOffset: Long,
-    brokerEpoch: Long
+    brokerEpoch: Long,
+    verifyBrokerEpoch: Boolean = false
```

Comment:
I would prefer not to use a default parameter. I guess we did it because `updateFetchStateOrThrow` has quite a few callers.

Maybe we can pass the value for `verifyBrokerEpoch` into the `Replica` constructor to minimize some of the changes? It also helps us keep the `updateFetchState` a bit simpler.

Reply:
Done.